File: tree-hugger/lib/tests/fixtures/signatures.php
<?php

/**
 * A standalone function with typed parameters and return type.
 */
function greet(string $name, int $age = 25): string {
    return "Hello, {$name}!";
}

/**
 * A variadic function.
 */
function greetMany(string ...$names): void {
    foreach ($names as $name) {
        echo "Hello, {$name}!\n";
    }
}

/**
 * A method inside a class.
 */
class Greeter {
    private string $prefix;

    public function __construct(string $prefix = "Hello") {
        $this->prefix = $prefix;
    }

    public function greet(string $name): string {
        return "{$this->prefix}, {$name}!";
    }
}

File: tree-hugger/lib/tests/fixtures/signatures.swift
import Foundation

/// A simple greeting function.
func greet(name: String) -> String {
    return "Hello, \(name)!"
}

/// A function with multiple parameters.
func greetWithAge(name: String, age: Int) -> String {
    return "Hello, \(name) (\(age))!"
}

/// A function with default parameter.
func greetWithPrefix(name: String, prefix: String = "Hello") -> String {
    return "\(prefix), \(name)!"
}

/// A void function.
func printGreeting(message: String) {
    print(message)
}

/// A function with variadic parameter.
func greetMany(names: String...) {
    for name in names {
        print("Hello, \(name)")
    }
}

/// A class with methods.
class Greeter {
    var prefix: String

    /// Initializer with parameter.
    init(prefix: String = "Hello") {
        self.prefix = prefix
    }

    /// A simple greeting method.
    func greet(name: String) -> String {
        return "\(prefix), \(name)!"
    }
}
